src/main/java/com/jwplayer/jwplatform/rest/HttpCalls.java

```diff
@@ -0,0 +1,101 @@
+			case "POST":
+				if (isBodyParams) {
+					requestUrl = buildRequestUrl(path, Collections.emptyMap());
+					response = Unirest.post(requestUrl).headers(headers).body(new JSONObject(params)).asJson();
+				} else {
+					requestUrl = buildRequestUrl(path, params);
+					response = Unirest.post(requestUrl).asJson();
+				}
+				break;
+			case "PATCH":
+				if (isBodyParams) {
+					requestUrl = buildRequestUrl(path, Collections.emptyMap());
+					response = Unirest.patch(requestUrl).headers(headers).body(new JSONObject(params)).asJson();
+				} else {
+					requestUrl = buildRequestUrl(path, params);
+					response = Unirest.patch(requestUrl).asJson();
+				}
+				break;
+			case "PUT":
+				if (isBodyParams) {
+					requestUrl = buildRequestUrl(path, Collections.emptyMap());
+					response = Unirest.put(requestUrl).headers(headers).body(new JSONObject(params)).asJson();
+				} else {
+					requestUrl = buildRequestUrl(path, params);
+					response = Unirest.put(requestUrl).asJson();
+				}
+				break;
+			case "DELETE":
+				response = Unirest.delete(path).headers(headers).asJson();
+				if(response.getStatus()== 201 || response.getStatus()== 200 || response.getStatus() == 204) return new JSONObject("{\"message\":\"Deletion successful\"}");
+				break;
+			default:
+				throw new JWPlatformException(String.format("%s is not a supported request type.", requestType));
+			}
+			final JSONObject responseBlock = response.getBody().getObject();
+			Util.checkForNon200Response(response.getStatus(), responseBlock);
+			return responseBlock;
+		} catch (final UnirestException e) {
+			throw new JWPlatformUnknownException(String.format("Non-JSON response from server: %s", e.toString()));
+		}
+	}
+	
+	/**
+	   * Generates fully formed URL for api request.
+	   *
+	   * @param path - endpoint to be used in API request which includes a leading slash (ie /my/path not my/path)
+	   * @param params - Parameters to be included in the request
+	   * @return - Fully formed request URL for an API request with api signature
+	   * @throws JWPlatformException - an exception occurred during encoding
+	   */
+	  private static String buildRequestUrl(
+	      final String path, final Map<String, String> params)
+	      throws JWPlatformException {
+	    final TreeMap<String, String> orderedParams = new TreeMap<>(params);
+	    orderedParams.put("api_format", "json");
+	    orderedParams.put("api_nonce", Util.getRandomNonce());
+	    orderedParams.put("api_timestamp", Util.getCurrentUnixTimestampInSeconds());
+
+	    final StringBuilder encodedParams = new StringBuilder();
+	    for (final String param : orderedParams.keySet()) {
+	      if (encodedParams.length() != 0) {
+	        encodedParams.append("&");
+	      }
+	      final String encodedValue = Util.encodeStringForJWPlatformAPI(orderedParams.get(param));
+	      encodedParams.append(param).append("=").append(encodedValue);
+	    }
+
+	    return path + "?" + encodedParams.toString();
+	  }
+	  
```

Comment:
extra new lines can be deleted. There is a checkstyle you can load into your IDE that will clean up some of this for you. https://github.com/jwplayer/jwplatform-java/blob/master/src/checkstyle/google_checks.xml